### PR TITLE
Cleanup of the rules depending on users

### PR DESCRIPTION
**Pull request by mbacovsky**
- missing groups were created in Rover
- users were added to the respective groups
- accesses for some legacy accounts were removed

### CODE REVIEW

**Comment by mbacovsky** [comment]
For some changes I'll ask @falox for double check.
As it may affect other please do not merge on Friday ;)

File: kfdefs/base/trino/trino-acl-rules.json [hunk 1]
```diff
@@ -28,20 +28,10 @@
         "owner": true
     },
     {
-        "user": "ccx-admin|ccx-reporting-pipeline-trino",
+        "group": "ccx-datalake-owners",
         "schema": "(ccx|ccx_sensitive|ccx_srep|ccx_internal|ccx_workloads)",
         "owner": true
     },
-    {
-        "group": "ccx-srep-data-access",
```
Comment (mbacovsky) [inline kfdefs/base/trino/trino-acl-rules.json:36]: I'm not sure why we gave ccx-srep-data-access and ccx-research-pipeline-trino ownership of the schemas. Is it needed to sync the partitions?

File: kfdefs/base/trino/trino-acl-rules.json [hunk 2]
```diff
@@ -144,45 +134,25 @@
...
+        "schema": "ccx_internal",
         "privileges": ["SELECT"]
     },
     {
         "group": "ccx-dev|assisted-lakers|ccx-datalake-access|ceeandpe|na-cs-tam-auto|apac-cs-tam-auto|latam-cs-tam-auto|emea-cs-tam-auto|na-ps-cs-tam-auto|emea-cs-csm-auto|emea-cs-cse-auto|emea-cs-managers|apac-cs-csm-auto|apac-cs-cse-auto|na-cs-csm-auto|na-cs-cse-auto|na-ps-cs-cse-auto|latam-cs-csm-auto|cs-csa-auto-ccx|ccx-pm|telemeter-auth|telemeter-auto-approval|telemeter-manual-approval|cee-sbr-shift|gcs-csm|asr-insights-dashboards",
         "schema": "ccx",
         "privileges": ["SELECT"]
     },
-    {
-        "user": "superset-ccx",
```
Comment (mbacovsky) [inline kfdefs/base/trino/trino-acl-rules.json:167]: @accorvin are these superset accounts still used? I didn't find them in Rover

File: kfdefs/base/trino/trino-acl-rules.json [hunk 3]
```diff
@@ -144,45 +134,25 @@
...
         "schema": "ccx_sensitive",
         "privileges": ["SELECT"]
     },
-    {
-        "user": "superset-ccx-sensitive",
-        "schema": "ccx_sensitive",
-        "privileges": ["SELECT"]
-    },
-    {
-        "user": "(telemetry-automated|telemetry-edmund-abbot|telemetry-analytics)",
```
Comment (mbacovsky) [inline kfdefs/base/trino/trino-acl-rules.json:182]: `telemetry-automated` is not in Rover neither Edmund Abbot so I skipped those. `telemetry-analytics` was added.

File: kfdefs/base/trino/trino-acl-rules.json (hunk 1 above)
Comment (accorvin) [reply kfdefs/base/trino/trino-acl-rules.json:36]: I'm not sure either. It seems safe to remove this to me.

**Comment by falox** [comment]
LGTM. I would merge next week (Mon 5th), unless @accorvin needs this week

**Comment by mbacovsky** [comment]
@accorvin what is the plan with this? Do you prefer to merge and test in Trino first or apply the changes to Starburst only? Is there anything I could do to help with either option?

**Comment by accorvin** [comment]
@mbacovsky hey, sorry, I forgot abut this one.

Everything looks good to me here. I'm inclined to merge this now and let it apply to trino, and I've already given the Starburst team the heads up to pull in these changes too.

Give me a final +1 and I'll merge this.

**Comment by mbacovsky** [comment]
Ok, lets merge. If we have Friday to monitor and fix eventual issues, +1 from me.

**Pull request merged**